### PR TITLE
Introducing request-level entities

### PR DESCRIPTION
**Pull request by kitchWWW**
Example usage:
```
      WitRequestOptions options = new WitRequestOptions();
      options.entityListProvider = new WitSimpleEntityList("colors", new List<string> {"red","aqua","forest"});
      wit.Activate("make the sphere aqua", options); 
```
Also works for speech activation.

### CODE REVIEW

File: Scripts/Runtime/WitRequestFactory.cs [hunk 1]
```diff
@@ -35,9 +42,14 @@ public static WitRequest MessageRequest(this WitConfiguration config, string que
         /// <param name="config"></param>
         /// <param name="maxBestIntents"></param>
         /// <returns></returns>
-        public static WitRequest SpeechRequest(this WitConfiguration config, int maxBestIntents = 1)
+        public static WitRequest SpeechRequest(this WitConfiguration config, WitRequestOptions requestOptions, int maxBestIntents = 1)
```
Comment (patapizza) [inline Scripts/Runtime/WitRequestFactory.cs:45]: let's move `maxBestIntents` to `WitRequestOptions`

File: Scripts/Runtime/Configuration/WitRequestOptions.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+using com.facebook.witai.interfaces;
+using com.facebook.witai.lib;
+using UnityEngine.Serialization;
+
+namespace com.facebook.witai
+{
+
+	public class WitRequestOptions
+    {
+        public IEntityListProvider  entityListProvider;
```
Comment (patapizza) [inline Scripts/Runtime/Configuration/WitRequestOptions.cs:24]: rm extra space

File: Scripts/Runtime/Configuration/WitRequestOptions.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+using System.Net;
+using com.facebook.witai.data;
+using com.facebook.witai.events;
+using com.facebook.witai.interfaces;
+using com.facebook.witai.lib;
+using UnityEngine.Serialization;
+
+namespace com.facebook.witai
+{
+
```
Comment (patapizza) [inline Scripts/Runtime/Configuration/WitRequestOptions.cs:21]: rm

File: Scripts/Runtime/VoiceService.cs [hunk 4]
```diff
@@ -48,6 +55,13 @@ public VoiceEvents VoiceEvents
         /// </summary>
         /// <param name="transcription"></param>
         public abstract void Activate(string transcription);
+        
+        /// <summary>
+        /// Send text data for NLU processing with custom request options.
+        /// </summary>
+        /// <param name="transcription"></param>
```
Comment (patapizza) [inline Scripts/Runtime/VoiceService.cs:62]: text

File: Scripts/Runtime/VoiceService.cs [hunk 5]
```diff
@@ -48,6 +55,13 @@ public VoiceEvents VoiceEvents
...
+        /// <param name="requestOptions"></param>
+        public abstract void Activate(string transcription, WitRequestOptions requestOptions);
```
Comment (patapizza) [inline Scripts/Runtime/VoiceService.cs:64]: s/transcription/text

File: Scripts/Runtime/VoiceService.cs [hunk 6]
```diff
@@ -70,7 +84,14 @@ public interface IVoiceService
         /// </summary>
         void Activate();
 
+        /// <summary>
+        /// Activate the microphone and send data for NLU processing with custom request options.
+        /// </summary>
+        /// <param name="requestOptions"></param>
+        void Activate(WitRequestOptions requestOptions);
```
Comment (patapizza) [inline Scripts/Runtime/VoiceService.cs:91]: can we provide this as default value for the parameter and dedup the methods?

File: Scripts/Runtime/WitRequestFactory.cs (hunk 1 above)
Comment (patapizza) [reply Scripts/Runtime/WitRequestFactory.cs:45]: ^

File: Scripts/Runtime/VoiceService.cs (hunk 5 above)
Comment (kitchWWW) [reply Scripts/Runtime/VoiceService.cs:64]: I was going off of the established parameter name, which was previously `transcription`. But this is a good change, I'll fix all of these instances.

File: Scripts/Runtime/VoiceService.cs (hunk 6 above)
Comment (kitchWWW) [reply Scripts/Runtime/VoiceService.cs:91]: Default values need to be compile-time constants. The default parameter such as `WitRequestOptions requestOptions = new WitRequestOptions()` isn't a compile-time constant, so was running into issues there. @yolanother do you know a better way to get around this?

File: Scripts/Runtime/WitRequestFactory.cs (hunk 1 above)
Comment (kitchWWW) [reply Scripts/Runtime/WitRequestFactory.cs:45]: done.

**Pull request merged**